src/Symfony/Component/Serializer/Normalizer/UidNormalizer.php

```diff
@@ -70,9 +69,15 @@ public function supportsNormalization($data, string $format = null)
     public function denormalize($data, string $type, string $format = null, array $context = [])
     {
         try {
-            return Ulid::class === $type ? Ulid::fromString($data) : Uuid::fromString($data);
+            return AbstractUid::class !== $type ? $type::fromString($data) : Uuid::fromString($data);
         } catch (\InvalidArgumentException $exception) {
             throw new NotNormalizableValueException(sprintf('The data is not a valid "%s" string representation.', $type));
+        } catch (\Error $e) {
+            if (str_starts_with($e->getMessage(), 'Cannot instantiate abstract class')) {
```

Comment:
Catching the error and looking for the message looks better than doing reflection to me, maybe there's a better way?